dali/ft/filecopy.cpp

```diff
@@ -3249,13 +3252,112 @@ void FileSprayer::updateTargetProperties()
                 if (stricmp(aname, "Protect") != 0)
                     curProps.addPropTree(aname, createPTreeFromIPT(&iter->query()));
             }
+
+            // Handle history
+            IPropertyTree &curHistory = lock.queryFile()->queryHistory();
+
+            // copy history records from source file
+            Owned<IPropertyTreeIterator> historyIter = srcHistory->getElements("*");
+            ForEach(*historyIter)
+            {
+                const char *aname = historyIter->query().queryName();
+                curHistory.addPropTree(aname, createPTreeFromIPT(&historyIter->query()));
+            }
+
+            // Add new record about this operation
+            Owned<IPropertyTree> newRecord = createPTree();
+
+            CDateTime temp;
+            temp.setNow();
+            unsigned hour, min, sec, nanosec;
+            temp.getTime(hour, min, sec, nanosec);
+            temp.setTime(hour, min, sec, 0);
+
+            StringBuffer timestr;
+            newRecord->addProp("@timestamp",temp.getString(timestr).str());
+
+            newRecord->addProp("@owner", srcAttr->queryProp("@owner"));
```

Comment:
As prev. mentioned attributes can't have multiple values, so setProp is more appropriate.